test/Flick/AnimatedCarousalPage.xaml.cs

```diff
@@ -71,42 +149,91 @@ protected override void OnNavigatedTo(NavigationEventArgs e)
+            scaleXExpression.SetReferenceParameter("item", item);
+
+            /* TODO: Expose ItemScaleRatio (scaleRatioXY) as a DependencyProperty in the custom Carousel
+             * control so the user can set it to any value */
+            scaleXExpression.SetScalarParameter("scaleRatioXY", (float)ItemScaleRatio);
+            scaleXExpression.SetScalarParameter("spacing", (float)layout.Spacing);
 
             // scale the item based on the distance of the item relative to the center of the viewport.
-            scaleExpresion.Expression = "1 - abs((svVisual.Size.X/2 - scrollProperties.Translation.X) - (item.Offset.X + item.Size.X/2))*(.75/(svVisual.Size.X/2))";
-            item.StartAnimation("Scale.X", scaleExpresion);
-            item.StartAnimation("Scale.Y", scaleExpresion);
+            //scaleExpression.Expression = "1 - abs((svVisual.Size.X/2 - scrollProperties.Translation.X) - (item.Offset.X + item.Size.X/2))*(.75/(svVisual.Size.X/2))";
+            var scaleExpressionString = "clamp((scaleRatioXY * (1 + (1 - (abs((item.Offset.X + (item.Size.X/2)) - ((svVisual.Size.X/2) - scrollProperties.Translation.X)) / (item.Size.X + spacing))))), scaleRatioXY, 1)";
+            scaleXExpression.Expression = scaleExpressionString;
 
-            var centerPointExpression = scrollProperties.Compositor.CreateExpressionAnimation();
-            centerPointExpression.SetReferenceParameter("item", item);
-            centerPointExpression.Expression = "Vector3(item.Size.X/2, item.Size.Y/2, 0)";
-            item.StartAnimation("CenterPoint", centerPointExpression);
-        }
+            // item.StartAnimation("Scale.X", scaleXExpression);
+            scaleXExpression.Target = "Scale.X";
+            animationGroup.Add(scaleXExpression);
 
-        private void OnItemGotFocus(object sender, RoutedEventArgs e)
-        {
-            ScrollToCenterOfViewport(sender);
+            var scaleYExpression = scrollProperties.Compositor.CreateExpressionAnimation();
+            scaleYExpression.SetReferenceParameter("svVisual", svVisual);
+            scaleYExpression.SetReferenceParameter("scrollProperties", scrollProperties);
+            scaleYExpression.SetReferenceParameter("item", item);
+
+            /* TODO: Expose ItemScaleRatio (scaleRatioXY) as a DependencyProperty in the custom Carousel
+             * control so the user can set it to any value */
+            scaleYExpression.SetScalarParameter("scaleRatioXY", (float)ItemScaleRatio);
+            scaleYExpression.SetScalarParameter("spacing", (float)layout.Spacing);
+
+            // scale the item based on the distance of the item relative to the center of the viewport.
+            scaleYExpression.Expression = scaleExpressionString;
+
+            // item.StartAnimation("Scale.Y", scaleXExpression);
+            scaleYExpression.Target = "Scale.Y";
+            animationGroup.Add(scaleYExpression);
```

Comment:
>scaleYExpression [](start = 31, length = 16)

can you just rename scaleXExpression as scaleExpression and use the same expression for Scale.X and scale.Y ? It looks like they are the same.

Reply:
Done.